### PR TITLE
[Merged by Bors] - Refactored the catalogs documentation

### PR DESCRIPTION
**Pull request by Jimvin**
# Description

Refactored the catalogs documentation and created a separate page for each supported Trino connector.

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by sbernauer — COMMENTED** [review]
Thx! Really like the section on connecting to existing Hive clusters!

File: docs/modules/usage_guide/pages/catalogs/hive.adoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      s3:
+        inline:
+          host: test-minio
+          port: 9000
+          accessStyle: Path
+          credentials:
+            secretClass: minio-credentials
+# We can use configOverrides to add arbitrary properties to the Trino catalog configuration
+  configOverrides:
+    hive.metastore.username: trino
```
[the remark this answers is not in the document]
Comment (Jimvin) [reply docs/modules/usage_guide/pages/catalogs/hive.adoc:34]: Good idea, I've added it there too.

File: docs/modules/usage_guide/pages/catalogs/index.adoc [hunk 2]
```diff
@@ -71,6 +77,6 @@ spec:
 ----
 
 The `spec.catalogLabelSelector` is used to fetch the list of `TrinoCatalogs` used for this Trino cluster.
-In this case the `hive` and `iceberg` catalogs will be used as they both match the `catalogLabelSelector`.
+In this case the `hive` and `iceberg` catalogs will be used as they both match the `catalogLabelSelector` value simple-trino    .
```
Comment (sbernauer) [inline docs/modules/usage_guide/pages/catalogs/index.adoc:80]:
```suggestion
In this case the `hive` and `iceberg` catalogs will be used as they both match the `catalogLabelSelector` value `simple-trino`.
```

**Review by sbernauer — APPROVED** [review]

**Pull request closed without merging**